pinot-controller/src/main/java/org/apache/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -135,6 +137,13 @@ protected void postprocess(Context context) {
     _controllerMetrics.setValueOfGlobalGauge(ControllerGauge.UPSERT_TABLE_COUNT, context._upsertTableCount);
     _controllerMetrics.setValueOfGlobalGauge(ControllerGauge.DISABLED_TABLE_COUNT, context._disabledTables.size());
 
+    // metric for total number of tables using a particular tier backend
+    context._tierBackendTableCountMap.forEach((tier, count) ->
+      _controllerMetrics.setOrUpdateGauge(tier + ControllerGauge.TIER_BACKEND_TABLE_COUNT.getGaugeName(), count));
```

Comment:
You also need to handle metrics deletion right?

Reply:
Good point! Tried to address it by maintaining the gauge names. Could not think of a cleaner way to handle it.